### DISCUSSION TITLE
Looking for a prices API for last financial year

### DISCUSSION THREAD

**Discussion by johnelliffe123** [amberelectric/public-api#164] [General]
Hi,
I am attempting to do an analysis of amber prices for the last financial year. I live in an apartment building where lifts are used at the beginning and end of each day. I interested in doing an analysis to determine whether it would be financially reasonable to purchase electricity from Amber in the middle of the day and early morning to charge a battery to power these lifts. 

The /sites/{siteId}/prices would be ideal for this purpose except that I have only been an Amber customer for 2 months and I would need to do an analysis over at least a 12 month period. Do you have an API which would give me prices for the ACT, independent of the site?

Thanks,
John

**Comment by madpilot** [comment]
We don't limit the time range—it should just work. Though maybe do it in 3-month tranches to avoid timeouts.

**Reply by madpilot** [reply]
And if it doesn't, write in to info@amber.com.au and ask - we can generate a CSV for you.

**Reply by johnelliffe123** [reply]
Thanks but my issue is that the /sites/{siteId}/prices api requires a siteId which is basically my account and I have only been a customer  for 2 months. This time next year I'll be able to get the data i am after.

**Reply by madpilot** [reply]
We only use the site information to build a tariff profile, but we don't limit the query to your start date - you can fetch prices back in time to 2018 (usually).

The reason we don't have a generic API for this is because setting up the tariff information is a bit painful.

If you want prices for a _different_ tariff - email in, and we can do that for you.

**Reply by timdcochran** [reply]
Is it possible this has changed? The following only returns price information since 24/12 this year, the same day my account was created:

% curl -X 'GET' \
  'https://api.amber.com.au/v1/sites/yippiedoo/prices?startDate=2022-12-01&endDate=2023-12-29&resolution=30' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer yaddayadda' -o kwh.json

**Reply by madpilot** [reply]
My apologies, I just checked the code, and it is limited to the account start date.

You'll need to email in to request the historic data.

**Reply by timdcochran** [reply]
Yeah all good @madpilot... I emailed info@amber.com.au as per your original email and they were back with the requested data within a day! Cheers

**Reply by serg3y** [reply]
I would also like historical data to perform analysis—primarily for SA, but other states too. Ideally, I would like to go as far back as possible to observe long-term trends.

I have been with Amber for three months and am very happy. I am also a strong believer in floating prices to encourage consumers to do the right thing, and profit from doing so.

I am trying to convince my skeptical friends that switching to Amber makes financial sense as well as being the responsible thing to do. They are very conservative and quote stories of one or two people getting burned more than four years ago. They require a lot of evidence/data to be convinced. I have been a customer for only three months, so I do not have enough 'evidence.'

I think it is in Amber's interest to be as transparent as possible and make all historical data:
1. Available to the public
2. Convenient to analyse

At the moment, users need to be customers and can only access data since the day they joined, and they also need to be able to get the API working. Note that the API documentation has mistakes in a few places, so it takes a bit of guesswork to get it working.

Suggestions:
1. One way to address all these issues is to make the data available by alternate means—maybe something as simple as a Google Drive link with an intuitive folder structure: `/prices/state/year/yyyymmdd.json`.
2. Build a page that displays all the data and lets users explore it in various way. This is a LOT of work, so I don't expect this any time soon. I suspect 3rd parties will build one for amber if data is made readily accessible by API.
3. Last but not least, fix the API documentation and make the prices endpoint and doc publicly accessible without requiring a token or `siteId`.

**Reply by rob0101** [reply]
_ I would also like historical data to perform analysis—primarily for SA_

We have an active Facebook Group for Amber SA prices.

https://www.facebook.com/groups/466861799632784

I've put a fair bit of data into the Files section to help people go back and see what was happening in previous months.

**Reply by madpilot** [reply]
> Note that the API documentation has mistakes in a few places, so it takes a bit of guesswork to get it working.

Got an example? We can fix that up

**Reply by serg3y** [reply]
Here are some suggestions to improve the API Doc.
I will use /prices endpoint as an example.
Some suggestions may not be practicable.
Not all are essential, I tried to give them relative priority using L/M/H = low/med/high.

Note: Currently workflow to test the /prices endpoint has about 13 steps, some of which are not obvious.
1. Click "Generate a new token"
2. Save the Token for later use
3. Click "Authorise"
4. Enter Token
5. Click "/sites" drop-down
6. Click "Try it out"
7. Click "Execute"
8. Copy "id" value from the reply
9. Click prices endpoint drop-down
10. Click "Try it out" 
11. Change default siteId
12. Change default time range
13. Click "Execute"

Suggested improvements:

1. L - Remove the "Try it out" button and have the UI start in the user input mode.
   Clicking the button does not seem to change anything.

2. M - Force user to click Authorise before they can click Execute.
   Grey out the "Execute" button and put red text next to it saying:
   "User must go to the top of the page and hit "Authorise" before running this test."

3. M - Improve text on the Authorise dialogue.
   Currently, it says "apiKey (http, Bearer)", which means little to most people.
   Suggested text "Enter a valid Token. Generate a new Token if you don't have one."

4. M - Use a better default time range.
   The current default time range is 2021-05-05 to 2021-05-05, which will return empty "[]" for any new user.
   Have the default time range start just a couple of days in the past.

5. H - Force user to enter a valid siteId.
   Remove the hard coded example siteId from the input dialogue. When users see this siteId, they think it is their personal siteId, since they have probably just completed the authorisation step.
   If possible, pre-fill with the user's actual siteId or leave it blank. If needed, provide an example in the text description above the input.
   ALSO, grey out the "Execute" button and add red text next to the button:
   "User must enter a valid siteId, which can be found using the /sites endpoint. Note the siteId is called "id" in the returned JSON."

6. H - Windows cmd.exe does not support single quotes (').
   The generated commands fail on Windows with error: "curl: (3) URL rejected: Port number was not a decimal number between 0 and 65535."
   Use double quotes (") to support Windows.

7. H - Windows cmd.exe does not support line breaks (\)
   The generated commands fail on Windows with error: "curl: (3) URL rejected: Bad hostname."
   Remove line breaks to support Windows.

8. L - Simplify the URL.
   This header does not seem to be required: -H "accept: application/json"

9. L - Tell users they cannot enter time, only date.

10. L - Change location and behaviour of URL and CURL command.
    Currently, the URL and CURL command become visible only after the user hits Execute.
    It would be informative to see them at the end of the parameter section and see them update on the fly as the user changes input values.
    Parameters
    ....
    URL:     https://api.amber.com.au/v1/sites/***/prices?startDate=2025-01-05&endDate=2025-01-05T11:30
    CURL:    curl -X GET "https://api.amber.com.au/v1/sites/***/prices?startDate=2025-01-05&endDate=2025-01-05T11:30" -H "Authorization: Bearer ***"
    [Execute]

Other UX improvements may be possible.
It may be possible to have just three steps to test an endpoint:
eg Click "Generate Token" > Select endpoint > Click "Test"
Where the website would automatically Authorise, pre-populate siteId and time ranges.

**Reply by serg3y** [reply]
> Got an example? We can fix that

Is anything in the list above actionable/useful?

**Reply by madpilot** [reply]
Yeah. A bunch of that stuff annoys me too. Perhaps I should search for a different OpenAPI viewer that has a better UX.

**Comment by serg3y** [comment]
Hello madpilot,

I am looking for more historic price data... just the spot price may be enough. Then I could calculate the various tariffs from that.

Is the spot price data already available elsewhere?
I tried AEMO > "aggregated-data", but it does not look like price data:
https://aemo.com.au/energy-systems/electricity/national-electricity-market-nem/data-nem/aggregated-data
REGION,SETTLEMENTDATE,TOTALDEMAND,RRP,PERIODTYPE
SA1,2025/01/01 00:05:00,1379.47,141.48,TRADE
SA1,2025/01/01 00:10:00,1367.61,136.59,TRADE
SA1,2025/01/01 00:15:00,1379.99,142.05,TRADE
...

Could the Amber API please expose the spot prices to the public?
It already does it for forecast data:
  https://api.amber.com.au/v1/state/vic/renewables/current?next=48&previous=48&resolution=30
**
Maybe something like this for historic spot price:
  https://api.amber.com.au/v1/state/vic/spot?startDate=2025-01-01&endDate=2025-01-02&resolution=30
**

PS,
I did join the facebook group, but I think the data there is limited.
I do have some Spot & RTOU prices via the API (5 months, as far as it lets me).
I requested and received 12 months of RTOU from amber, by email, but no spot price, and SA only.

**Reply by madpilot** [reply]
It's unlikely we'll do that, but you can use OpenElectricity: https://openelectricity.org.au

FYI the RRP field of that AEMO feed is the spot price (ex-GST)

We're not able to price sites prior to a user becoming a customer due to limitations in our pricer (it doesn't know what tariff you are on before joining).
